internal/bgp/bgp.go

```diff
@@ -427,3 +447,211 @@ func (a *Advertisement) Equal(b *Advertisement) bool {
 	}
 	return reflect.DeepEqual(a.Communities, b.Communities)
 }
+
+const (
+	//tcpMD5SIG TCP MD5 Signature (RFC2385)
+	tcpMD5SIG = 14
+	//ipv6MINHOPCOUNT Generalized TTL Security Mechanism (RFC5082)
+	ipv6MINHOPCOUNT = 73
+)
+
+// This  struct is defined at; linux-kernel: include/uapi/linux/tcp.h,
+// It  must be kept in sync with that definition, see current version:
+// https://github.com/torvalds/linux/blob/v4.16/include/uapi/linux/tcp.h#L253
+type tcpmd5sig struct {
+	ssFamily uint16
+	ss       [126]byte
+	pad1     uint16
+	keylen   uint16
+	pad2     uint32
+	key      [80]byte
+}
+
+func buildTCPMD5Sig(address string, key string) (tcpmd5sig, error) {
+	t := tcpmd5sig{}
+	addr := net.ParseIP(address)
+	if addr.To4() != nil {
+		t.ssFamily = syscall.AF_INET
+		copy(t.ss[2:], addr.To4())
+	} else {
+		t.ssFamily = syscall.AF_INET6
+		copy(t.ss[6:], addr.To16())
+	}
+
+	t.keylen = uint16(len(key))
+	copy(t.key[0:], []byte(key))
+
+	return t, nil
+}
+
+//TCPDialer  represents the connection
+type TCPDialer struct {
+	net.Dialer
+
+	// MD5 authentication password.
+	AuthPassword string
+
+	// The TTL value to set outgoing connection.
+	TTL uint8
```

Comment:
This option doesn't seem related to the TCP-MD5 change, and setting the TTL to a static value different from the system default can lead to surprising behavior. I would suggest just removing this, unless you have a specific need for it.

In general, I try to keep "generic" options and tweaks out of MetalLB unless someone demonstrates a specific need for it, to limit the additional complexity in the code, and the number of different combinations that have to be tested.